src/shared/components/pipeline-run-logs/logs/LogsWrapperComponent.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import { useK8sWatchResource, WatchK8sResource } from '../../../../dynamic-plugin-sdk';
+import { LoadingBox } from '../../status-box/StatusBox';
+import { PodKind } from '../../types';
+import { MultiStreamLogs } from './MultiStreamLogs';
+
+type LogsWrapperComponentProps = {
+  taskName: string;
+  downloadAllLabel?: string;
+  onDownloadAll?: () => Promise<Error>;
+  resource: WatchK8sResource;
+};
+
+const LogsWrapperComponent: React.FC<LogsWrapperComponentProps> = ({ resource, ...props }) => {
+  const resourceRef = React.useRef(null);
+  const [obj, loaded, error] = useK8sWatchResource<PodKind>(resource);
+
+  if (loaded && !error && resource.name === obj.metadata.name) {
+    resourceRef.current = obj;
+  }
+
+  return resourceRef.current ? (
+    <MultiStreamLogs {...props} resourceName={resource.name} resource={resourceRef.current} />
+  ) : (
+    <LoadingBox />
+  );
```

Comment:
@rohitkrai03 Loading of the pod is a bit slow because of the polling we do in useK8sWatchResource. when a task is selected new pod is fetched but useK8sWatchResource sends updates of the previously fetched resource that causes fluctuation on UI between the current task and previously selected task. Hence this logic was added to keep the pod resource consistent and to Send resource updates to MultiStreamLog component only when the `resource.name` matches the result from useK8sWatchResource.
![Kapture 2022-03-11 at 23 00 31](https://user-images.githubusercontent.com/9278015/157922709-387d7361-e027-4042-afe6-56a1fca2d96d.gif)